src/structure.cpp

```diff
@@ -91,11 +91,13 @@ static binary_partition *split_by_cost(int n, grid_volume gvol, bool fragment_co
       (gvol.surroundings().in_direction(best_split_direction) * best_split_point) /
       gvol.num_direction(best_split_direction);
     binary_partition *bp_split = new binary_partition(best_split_direction, best_split_position);
-    grid_volume left_gvol = gvol.split_at_fraction(false, best_split_point, best_split_direction);
     const int num_left = (size_t)(left_effort_fraction * n + 0.5);
-    bp_split->left = split_by_cost(num_left, left_gvol, fragment_cost);
-    grid_volume right_gvol = gvol.split_at_fraction(true, best_split_point, best_split_direction);
-    bp_split->right = split_by_cost(n - num_left, right_gvol, fragment_cost);
+    if (num_left > 0) {
```

Comment:
Should we be checking `num_left > 0 && num_left < n` here, to ensure that both the left and right trees are non-empty?

Reply:
Done.